packages/tc-schema-sdk/data-accessors/relationship-type.js

```diff
@@ -60,6 +62,17 @@ const formatSimpleRelationship = (
 	};
 };
 
+const formatRichRelationship = richRelationshipDefinition => {
```

Comment:
I've left some comments on the implementation, but I think this whole function could do with a rename and complete rewrite.

name: something like `setOtherNodeName`

implementation: I guess maybe part of the reason for implementing like this is because the objects are frozen so can't set a property on them? But even so, I think it'd help to check early  `if(from.otherNodeName && to.otherNodeName)` and just return richRelationshipDefinition without modification if it is. Then only carry out the thing you've implemented if necessary. This makes it far clearer that otherNodeName is sometimes defined already